packages/features/bookings/lib/useCheckOverlapWithOverlay.tsx

```diff
@@ -0,0 +1,35 @@
+import type { Dayjs } from "@calcom/dayjs";
+import dayjs from "@calcom/dayjs";
+
+import { useOverlayCalendarStore } from "../Booker/components/OverlayCalendar/store";
+
+function getCurrentTime(date: Date) {
+  const hours = date.getHours().toString().padStart(2, "0");
+  const minutes = date.getMinutes().toString().padStart(2, "0");
+  return `${hours}:${minutes}`;
+}
+
+export function useCheckOverlapWithOverlay(start: Dayjs, selectedDuration: number | null) {
+  const overlayBusyDates = useOverlayCalendarStore((state) => state.overlayBusyDates);
+
+  const overlappingTimeStart: string | null = null;
+  const overlappingTimeEnd: string | null = null;
+
+  const isOverlapping =
+    overlayBusyDates &&
+    overlayBusyDates.some((busyDate) => {
```

Comment:
I experienced some issues with timezones. 

I have a google calendar event in Amsterdam timezone and it correctly shows the slot as overlapping when I have Amsterdam timezone on the bookings page set. But when I change the booking page to London timezone, no slot is marked as overlapping anymore

Reply:
Think i have fixed this and also https://github.com/calcom/cal.com/pull/11693#discussion_r1350147627

Let me know if you run into any more issues